src/ledger-bridge-keyring.test.ts

```diff
@@ -0,0 +1,866 @@
+const fakeHdKey = HDKey.fromExtendedKey(fakeXPubKey);
+const fakeTx = new EthereumTx({
+  nonce: '0x00',
+  gasPrice: '0x09184e72a000',
+  gasLimit: '0x2710',
+  to: '0x0000000000000000000000000000000000000000',
+  value: '0x00',
+  data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057',
+  // EIP 155 chainId - mainnet: 1, ropsten: 3
+  chainId: 1,
+});
+
+const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.Berlin });
+const commonEIP1559 = new Common({
+  chain: Chain.Mainnet,
+  hardfork: Hardfork.London,
+});
+const newFakeTx = TransactionFactory.fromTxData(
+  {
+    nonce: '0x00',
+    gasPrice: '0x09184e72a000',
+    gasLimit: '0x2710',
+    to: '0x0000000000000000000000000000000000000000',
+    value: '0x00',
+    data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057',
+  },
+  { common, freeze: false },
+);
+
+const fakeTypeTwoTx = TransactionFactory.fromTxData(
+  {
+    nonce: '0x00',
+    maxFeePerGas: '0x19184e72a000',
+    maxPriorityFeePerGas: '0x09184e72a000',
+    gasLimit: '0x2710',
+    to: '0x0000000000000000000000000000000000000000',
+    value: '0x00',
+    data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057',
+    type: 2,
+    v: '0x01',
+  },
+  { common: commonEIP1559, freeze: false },
+);
+
+chai.use(spies);
+
+/**
+ * Checks if the iframe provided has a valid contentWindow
+ * and onload function.
+ *
+ * @param iframe - The iframe to check.
+ * @returns Returns true if the iframe is valid, false otherwise.
+ */
+function isIFrameValid(
+  iframe?: HTMLIFrameElement,
+): iframe is HTMLIFrameElement & { contentWindow: Window } & {
+  onload: () => any;
+} {
+  return (
+    iframe !== undefined &&
+    hasProperty(iframe, 'contentWindow') &&
+    typeof iframe.onload === 'function' &&
+    hasProperty(iframe.contentWindow as Window, 'postMessage')
+  );
+}
+
+/**
+ * Simulates the loading of an iframe by calling the onload function.
+ *
+ * @param iframe - The iframe to simulate the loading of.
+ * @returns Returns a promise that resolves when the onload function is called.
+ */
+async function simulateIFrameLoad(iframe?: HTMLIFrameElement) {
+  if (!isIFrameValid(iframe)) {
+    throw new Error('the iframe is not valid');
+  }
+  // we call manually the onload event to simulate the iframe loading
+  return await iframe.onload();
+}
+
+describe('LedgerBridgeKeyring', function () {
+  let keyring: LedgerBridgeKeyring;
+  let sandbox: ChaiSpies.Sandbox;
+
+  /**
+   * Sets up the keyring to unlock one account.
+   *
+   * @param accountIndex - The index of the account to unlock.
+   * @returns Returns a promise that resolves when the keyring is unlocked.
+   */
+  async function basicSetupToUnlockOneAccount(accountIndex = 0) {
+    keyring.setAccountToUnlock(accountIndex);
+    await keyring.addAccounts();
+    sandbox.on(keyring, 'unlock', async () =>
+      Promise.resolve(fakeAccounts[accountIndex]),
```

Comment:
Nit: Redundant Promise

```suggestion
      fakeAccounts[accountIndex],
```